src/contracts/NFTFactory.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract NFTFactory is ERC721Enumerable, Ownable {
+    string public baseURI;
+    uint256 public cost = 10 wei;
+
+    event Log(address sender, string message);
+    event Received(address caller, uint amount, string message);
+
+    constructor(
+        string memory _name,
+        string memory _token
```

Comment:
I guess this should be `symbol` instead of `token`. `token` variable name is a bit confusing.